### PR TITLE
[documentation] template for new component issue

### PR DESCRIPTION
**Pull request by codeboten**
Adding a template to use as an experiment in this repository. The workflow is expecting users to open a new component issue when proposing an issue, to reduce unnecessary work both for contributors and maintainers.

Once the template is approved, it can be copied over to the core repository and linked from the docs as well.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/new_component.md
Comment (bogdandrutu) [inline .github/ISSUE_TEMPLATE/new_component.md:38]: This is done automatically (when you link to `internal/components`), they need only the replace statement correct?

File: .github/ISSUE_TEMPLATE/new_component.md
Comment (bogdandrutu) [inline .github/ISSUE_TEMPLATE/new_component.md:40]: Is this something decided in the SIG meeting to not be a step, since we accept one large PR? I am trying to say that this is normal, not sure we need to link it here as a step.

File: .github/ISSUE_TEMPLATE/new_component.md
Comment (bogdandrutu) [inline .github/ISSUE_TEMPLATE/new_component.md:28]: Should we sort it somehow?

File: .github/ISSUE_TEMPLATE/new_component.md
Comment (bogdandrutu) [inline .github/ISSUE_TEMPLATE/new_component.md:33]: Is this not what we should ask in the issue template, since the first PR from the current process is no longer required?

File: .github/ISSUE_TEMPLATE/new_component.md
Comment (bogdandrutu) [inline .github/ISSUE_TEMPLATE/new_component.md:34]: Same, this is the example config from your template? when they write the code they definitely need a config. Maybe a separate section we can explain how to structure a component: config (embed the settings, bla bla); factory (use the helpers, bla bla), etc.

**Comment by bogdandrutu** [comment]
@codeboten see also https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components

**Comment by tigrannajaryan** [comment]
Do we need this? Is there any evidence that users often create issues of this kind? Check existing issues perhaps to show that this is a top request type?

**Comment by codeboten** [comment]
> Do we need this? Is there any evidence that users often create issues of this kind? Check existing issues perhaps to show that this is a top request type?

This was proposed in the SIG meeting on Jan 5 to reduce friction for reviewers and contributors. The thinking here is that this would replace the need to have a first PR when adding a component. Instead the issue containing the same level of information as that initial PR would suffice.

It's related to https://github.com/open-telemetry/opentelemetry-collector/issues/4630

File: .github/ISSUE_TEMPLATE/new_component.md
Comment (jpkrohling) [reply .github/ISSUE_TEMPLATE/new_component.md:33]: +1, the original "checklist" mentioned a readme file with these two items, but they should now be included in the issue, as there's no initial PR.

**Comment by jpkrohling** [comment]
I just updated #6926 with a comment based on my recollection of what we agreed during the latest SIG call: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6926#issuecomment-1009153700

The major difference is that it just mentioned the issue that has to be created, what the initial PR should contain, and the step about the -releases repo. I think we might then need two templates, one for the initial issue and one for the PR.

File: .github/ISSUE_TEMPLATE/new_component.md
Comment (mx-psi) [inline .github/ISSUE_TEMPLATE/new_component.md:41]: This may be confusing to people ("what should the Makefile contents be?") so maybe we can clarify this (e.g. "copy from template here")

**Comment by codeboten** [comment]
Removed the checklist, which can be added as a PR checklist. Updated the template with some links to example content and updated the contributing document to point to creating a new issue. @bogdandrutu @jpkrohling @mx-psi @tigrannajaryan PTAL

**Review by jpkrohling — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

File: CONTRIBUTING.md
Comment (tigrannajaryan) [inline CONTRIBUTING.md:11]: Is the sponsor only responsible for the review or also for subsequent maintenance?

Comment (codeboten) [reply CONTRIBUTING.md]: Thanks for the callout, i've added a note about the sponsor becoming a code owner as well.

File: CONTRIBUTING.md
Comment (codeboten) [reply CONTRIBUTING.md]: Fixed, i fixed the formatting earlier but hadn't saved it before submitting the change.

File: .github/ISSUE_TEMPLATE/new_component.md
Comment (JamieDanielson) [inline .github/ISSUE_TEMPLATE/new_component.md:12]:
```suggestion
<!--
*This information can be used later on to populate the README for the component. See an example overview [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/awsecscontainermetricsreceiver#overview).*
-->
```

Any reason to not make these commented blocks in the issue template? Might help ensure submitters keep the header (consider making a `##`) and remove this block to reduce noice in submission

File: CONTRIBUTING.md
Comment (jpkrohling) [reply CONTRIBUTING.md:11]: Sometimes, the idea for a component circulates via Slack first. But I agree that the typical case would probably be that sponsors have to be found after the idea is exposed via the initial issue.

So, add a new statement to this phrase, like:

> If there are no sponsors yet for the component, it's fine: use the issue as a means to try to find a sponsor for your component.

File: .github/ISSUE_TEMPLATE/new_component.md
Comment (codeboten) [reply .github/ISSUE_TEMPLATE/new_component.md]: Good call, done!

File: CONTRIBUTING.md
Comment (codeboten) [reply CONTRIBUTING.md:11]: Added.

**Review by jpkrohling — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

File: CONTRIBUTING.md
Comment (djaglowski) [inline CONTRIBUTING.md:13]:
```suggestion
* Some information about your component, such as the reasoning behind it, use-cases, telemetry data types supported, and anything else you think is relevant for us to make a decision about accepting the component.
* The configuration options your component will accept. This will help us understand what it does and have an idea of how the implementation might look like.
```

File: CONTRIBUTING.md
Comment (djaglowski) [inline CONTRIBUTING.md:12]: > for us to make a decision about accepting the component

Once the issue is opened, what is the process for making a decision? How is the decision indicated?

Comment (codeboten) [reply CONTRIBUTING.md]: What are your thoughts? The best I can think of is approvers/maintainers commenting on the issue that a decision has been made to approve/reject a proposal.

Comment (bogdandrutu) [reply CONTRIBUTING.md]: Labels? "Proposed Component", "Approved Component", and after X weeks close the issue as no decision made => no interest in supporting this?

Comment (djaglowski) [reply CONTRIBUTING.md]: > Labels? "Proposed Component", "Approved Component", and after X weeks close the issue as no decision made => no interest in supporting this?

I like this. The "Sponsor Needed" label can be applied until a decision is made.

**Pull request merged**
